### PR TITLE
Info icon tooltips on plan details page

### PR DESCRIPTION
**Pull request by petsch0**
implements #954 

9d2909df-0681-48ff-a897-495056c78f5d (4x)

### CODE REVIEW

**Comment by petsch0** [comment]
@seppeljordan @sloschert since I am not an expert on html and templating, kindly have a look at the initial commit and let me know what you think. This dropdown-thing looks a little hacky to me, but it is comfortable in the UI. Still I think that it would also be okay to re-use the existing info modal from draft_form.html.

**Comment by sloschert** [comment]
I like it, but it seems not to be mobile responsive, because the info text is cut off. Do you think there is a way to make the text fit mobile screens? 

Also, wouldn't be "tooltip" a more accurate name than "dropdown"?

image

**Review by sloschert — CHANGES_REQUESTED** [review]
I left one comment

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (petsch0) [reply arbeitszeit_flask/templates/macros/plan_details.html:15]: np, thanks for having a look. I changed the text according to your suggestion.

**Comment by petsch0** [comment]
> Do you think there is a way to make the text fit mobile screens?
thanks for checking this. I'll have a look to see if that is possible.

> Also, wouldn't be "tooltip" a more accurate name than "dropdown"?
yes, I changed the PR title, and will also change the file name and the macro.

**Comment by sloschert** [comment]
In thought it would be possible to tab tooltips on mobile, but might be wrong. Sure, we can do it in a future PR (but some effort has been made yet to achieve responsiveness, so we should not forget this). I'll approve and leave it up to @petsch0 how to proceed.

**Review by sloschert — APPROVED** [review]

**Comment by petsch0** [comment]
Consider latest texts for various fields as drafts for discussion.

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (petsch0) [inline arbeitszeit_flask/templates/macros/plan_details.html:66]: can plan be approved but be Inactive and not expired because current date is BEFORE start of planning timeframe?

**Comment by petsch0** [comment]
also as texts are getting longer and longer: do you think existing modals with pop-up windows should rather be re-used instead of those long dropdowns?

**Comment by sloschert** [comment]
> I see that approval is still valid although I pushed some commits. Shouldn't this have been invalidated, maybe with exception of merge commits? 

Apparently we could activate this option in github: https://stackoverflow.com/a/69229270 

but if we activate it we could not allow small cosmetic changes after approval (sometimes we say: "ok change this comment and you can merge"). For now I propose that you simply rerequest a review from us manually.

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (sloschert) [reply arbeitszeit_flask/templates/macros/plan_details.html:66]: No, currently not.

**Comment by sloschert** [comment]
> also as texts are getting longer and longer: do you think existing modals with pop-up windows should rather be re-used instead of those long dropdowns?

My opinion: I like the popups you implemented more, because they are less intrusive, and I would actually replace the existing modals with these. Regarding the length of the text: I think these tips should be quite short anyway. If they are too long to fit in such a info popup we should rather add this info to a proper user manual (on which we still have to agree, there is a rudimentary one in route `/help`).

File: arbeitszeit_flask/templates/macros/tooltips.html
Comment (sloschert) [inline arbeitszeit_flask/templates/macros/tooltips.html:14]: I find the text a bit too small - what about adding the class "is-size-6"?

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (sloschert) [inline arbeitszeit_flask/templates/macros/plan_details.html:133]: I think I would propose to delete "that can vay from plan to plan", because also fixed means of production can vary from plan to plan (not every machine will be used for every product). 
 Perhaps we should mention that we refer to marxian "circulating means of production" here? Perhaps we should even rename "liquid" to "circulating". See the wiki page: https://en.wikipedia.org/wiki/Circulating_capital. Not quite sure why we chose "liquid" from the beginning.  

 See Grundprinzipien 1930:
 "As is well known the Marxist category "means of production" comprises machinery and buildings (fixed means of production), and also raw materials and auxiliary materials (circulating means of production). If now we use for fixed means of production the letter p and for the circulating means of production the letter c, then the formula takes on the following form: ( p + c ) + L = product" https://www.marxists.org/subject/left-wing/gik/1930/03.htm
 In German: "Wie bekannt, umfaßt die Marx'sche Kategorie "Produktionsmittel" , Maschinen, Gebäude (feste Produktionsmittel), als auch die Rohmaterialien und Hilfsstoffe (zirkulierende Produktionsmittel). "

 In Grundprinzipien 1935:
 "Each company calculates how much working time is spent on its product, i.e., it determines how many working hours of "fixed" means of production (machines and buildings), how many working hours of "circulating" means of production (raw materials and consumables) and how many directly consumed working hours flow into the product." (p. 111f., https://arbeitszeit.noblogs.org/files/2023/04/GIC-Fundamental-Principles-2.-Ed.1935-1.pdf)

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (sloschert) [inline arbeitszeit_flask/templates/macros/plan_details.html:156]: Without "level"?

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (sloschert) [inline arbeitszeit_flask/templates/macros/plan_details.html:180]: 1) typo: "is THE account"
2) Actually there are only two reasons why the "price" differs from the costs: Either the plan is part of what we call a "price cooperation" [Grundprinzipien 1930, p. 101: "Die horizontale Zusammenfassung ist zunächst nur eine rechnerische Angelegenheit zur Feststellung der gesellschaftlich durchschnittlichen Produktionszeit und im Zusammenhang damit des Grades der Produktivität der einzelnen Betriebe. Sicher kann und darf es dabei nicht bleiben, sondern es muß auch zu gegenseitiger technischer Durchdringung kommen."] OR the plan is a public plan, then it is zero.
As the info text is now, it appears as if there could be several, and arbitary, reasons why the "price could be reduced".

**Comment by sloschert** [comment]
@petsch0 I left some comments on the info texts for your consideration and perhaps for some future discussions. I think they are a great feature.

File: arbeitszeit_flask/templates/macros/tooltips.html
Comment (petsch0) [reply arbeitszeit_flask/templates/macros/tooltips.html]: done

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (petsch0) [reply arbeitszeit_flask/templates/macros/plan_details.html]: done

File: arbeitszeit_flask/templates/macros/plan_details.html
Comment (petsch0) [reply arbeitszeit_flask/templates/macros/plan_details.html]: done

**Review by sloschert — APPROVED** [review]
Thanks! :-) 4xCR

**Pull request merged**
